docs/guides/nwaku/configure-nwaku.md

```diff
@@ -0,0 +1,246 @@
+---
+title: Nwaku Configuration Examples
+---
+
+This guide provides detailed steps to configure a `nwaku` node for different use cases.
+
+## Connect to Other Peers
+
+To join the Waku Network, nodes must [bootstrap](/overview/reference/glossary#bootstrapping) for an entry point before discovering more peers. Nwaku provides multiple [peer discovery](/overview/concepts/peer-discovery) mechanisms:
+
+- [Configure Static Peers](/guides/nwaku/configure-discovery#configure-static-peers)
+- [Configure DNS Discovery](/guides/nwaku/configure-discovery#configure-dns-discovery)
+- [Configure Discv5](/guides/nwaku/configure-discovery#configure-discv5)
+- [Configure Peer Exchange](/guides/nwaku/configure-discovery#configure-peer-exchange)
+
+## Configure a Domain Name
+
+You can set up an IPv4 DNS domain name that resolves to the public IPv4 address of a node using the `dns4-domain-name` option. This allows the node's publicly announced multiaddrs to use the `/dns4` scheme.
+
+```bash
+./build/wakunode2 --dns4-domain-name=[DOMAIN NAME]
+```
+
+For example, consider the domain name `nwakunode.com`, which resolves to a `nwaku` node:
+
+```bash
+./build/wakunode2 --dns4-domain-name=nwakunode.com
+```
+
+Nodes with a domain name and secure WebSocket configured will generate a discoverable ENR with `/wss` multiaddr and `/dns4` domain name, essential for verifying domain certificates when connecting securely.
+
+:::info
+This example describes configuring a domain name that resolves to your node's IP address and is unrelated to [DNS Discovery](/overview/concepts/dns-discovery).
+:::
+
+## Configure Store Protocol and Message Store
+
+To enable the [Store protocol](/overview/concepts/protocols#store) in `nwaku`, use the following configuration options:
+
+- `store`: Enables the `Store` protocol on the node (disabled by default).
+- `store-message-retention-policy`: Retention policy of the store node (how long messages will be persisted). Two different retention policies are supported:
+	- Time retention policy: `time:<duration-in-seconds>` (e.g., `time:14400`)
+	- Capacity retention policy: `capacity:<messages-count>` (e.g, `capacity:25000`)
+	- Set this option to an empty string to disable the retention policy.
+- `store-message-db-url`: Database connection URL for persisting messages in the [SQLAlchemy database URL format](https://docs.sqlalchemy.org/en/20/core/engines.html#database-urls). Setting this option to an empty string will instruct the node to use the fallback in-memory message store.
```

Comment:
Would be good to clarify whether these arguments are optional and if not set, what are the default behaviours